src/client/components/RegistrationMarketingConsentFormField.tsx

```diff
@@ -1,40 +1,20 @@
-import { css } from '@emotion/react';
-import { palette, space, textSans } from '@guardian/source/foundations';
 import React from 'react';
 import { ToggleSwitchInput } from '@/client/components/ToggleSwitchInput';
 
-const switchRow = css`
-	border: 0;
-	padding: 0;
-	margin: 0;
-	${textSans.medium()}
-	border-radius: 4px;
-	border: 1px solid ${palette.neutral[38]};
-	padding: ${space[2]}px;
-`;
-
-const labelStyles = css`
-	justify-content: space-between;
-	color: ${palette.neutral[46]};
-	${textSans.xsmall()}
-`;
-
 type Props = {
 	id: string;
-	label: string;
+	description: string;
 };
 
-export const RegistrationMarketingConsentFormField = ({ id, label }: Props) => {
+export const RegistrationMarketingConsentFormField = ({
```

Comment:
The `RegistrationMarketingConsentFormField` and `RegistrationNewsletterFormField` are looking basically like the same component with some slightly different props. Could they be combined somehow?

You can always use type narrowing so that depending on the combination of props you provide it will narrow to a specific type.

So it would look something close to this in a single component?

```ts
type MarketingProps = {
	id: string;
	description: string;
  title: undefined;
  imagePath: undefined;
  defaultChecked?: boolean;
};

type NewsletterProps = {
	id: string;
	title: string;
	description: string;
	imagePath?: string;
	defaultChecked?: boolean;
};

type Props = MarketingProps | NewsletterProps

export const RegistrationConsentFormField = ({
  description,
  id,
  title,
  imagePath,
  defaultChecked,
}: Props) => {
  return (
		<ToggleSwitchInput
			id={id}
			title={title}
			defaultChecked={defaultChecked}
			description={description}
			imagePath={imagePath}
		/>
	);
}
```

Reply:
Just realised we don't even need this component at all - the wrapper doesn't add anything useful. Removing it entirely.